core/vm/logger.go

```diff
@@ -65,36 +58,34 @@ type StructLog struct {
 	Err     error
 }
 
+type VmLogger interface {
+	CaptureState(env Environment, pc uint64, op OpCode, gas, cost *big.Int, memory *Memory, stack *stack, contract *Contract, depth int, err error);
+}
```

Comment:
I named it that because of this comment still attached to Logger, a few lines below:

> // Logger is an EVM state logger and implements VmLogger.

`StructLogger` definitely seems unsuitable, since it explicitly doesn't log structs. :)